### PR TITLE
Adds more shields and tweaks a few!

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

2 new shields that are weaker in melee but stand out in were they are needed
Laser Resistant shield is weak to bullets and the like only able to take 1-2 before shattering well being able to first off block(!) lasers as well as take more abuse from them then normal
Bullet Resistant shield is like the riot shield were lasers go right through but can withstand just a few more shots from a gun, sadly is weaker to melee making it not an upgrade!
Adds in Shield implant to the Weapon Implant node
Changes the needed mat to fix wooden shields to wood well keeping riots/towers needing titanium 

## Why It's Good For The Game

Makes more types of shields to think about and want in case of different problems.
AP ammo still kills the bullet resistant shields just as a like armor well AP will be stopped in its tracks from killing the shield. 
Well the laser shield can take lasers, the bullet shield can not, meaning its mainly used against bullets and some melee unlike the laser shield. Riot shields are still better at melee then bullet shields, not a flat upgrade!

## Changelog
:cl:
add: New shields have been added into the game Bullet Resistant and Laser Resistant shields.
add: New implanted shield can be gotten as well! - Has many drawbacks but can save your life!
balance: Wooden base shields now are fixed with wood rather then titanium 
imageadd: Moves all shields from items and weapons to a new file - shields
/:cl:

### CODE REVIEW

File: code/game/objects/items/shields.dm [hunk 1]
```diff
@@ -240,6 +209,12 @@
 	item_flags = SLOWS_WHILE_IN_HAND
 	transparent = FALSE
 
+/obj/item/shield/riot/tower/swat
+	name = "swat shield"
+	desc = "A massive, heavy shield that can block a lot of attacks, can take a lot of abuse before braking."
```
Comment (silicons) [inline code/game/objects/items/shields.dm]: breaking

File: code/game/objects/items/shields.dm [hunk 2]
```diff
@@ -23,6 +23,7 @@
 	custom_materials = list(/datum/material/glass=7500, /datum/material/iron=1000)
 	attack_verb = list("shoved", "bashed")
 	var/cooldown = 0 //shield bash cooldown. based on world.time
+	var/fixing = /obj/item/stack/sheet/mineral/titanium
```
Comment (silicons) [inline code/game/objects/items/shields.dm]: name this something better like repair_material or something

File: code/game/objects/items/shields.dm [hunk 3]
```diff
@@ -255,3 +230,55 @@
...
+	var/base_icon_state = "eshield" // [base_icon_state]1 for expanded, [base_icon_state]0 for contracted
+	var/on_force = 10
+	var/on_throwforce = 8
+	var/on_throw_speed = 2
+	var/active = 0
+	var/clumsy_check = TRUE
+
+/obj/item/shield/energy/Initialize()
+	. = ..()
+	icon_state = "[base_icon_state]0"
```
Comment (silicons) [inline code/game/objects/items/shields.dm:395]: all icon_state changes should be in update_icon_state(). make this default to 0 don't make an initialize override over this. make update icon called during toggling or something.

**Review by silicons — CHANGES_REQUESTED** [review]
1

File: code/game/objects/items/shields.dm (hunk 3 above)
Comment (Trilbyspaceclone) [reply code/game/objects/items/shields.dm:395]: Oh this wasnt me. I just moved the E-shield down to its own thing...

**Review by silicons — APPROVED** [review]

**Pull request merged**